lib/rouge/lexers/ghc_cmm.rb

```diff
@@ -0,0 +1,340 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+# C minus minus (Cmm) is a pun on the name C++. It's an intermediate language
+# of the Glasgow Haskell Compiler (GHC) that is very similar to C, but with
+# many features missing and some special constructs.
+#
+# Cmm is a dialect of C--. The goal of this lexer is to use what GHC produces
+# and parses (Cmm); C-- itself is not supported.
+#
+# https://gitlab.haskell.org/ghc/ghc/wikis/commentary/compiler/cmm-syntax
+#
+module Rouge
+  module Lexers
+    class GHCCmm < RegexLexer
+      title "GHC Cmm (C--)"
+      desc "GHC Cmm is the intermediate representation of the GHC Haskell compiler"
+      tag 'ghc-cmm'
+      filenames '*.cmm', '*.dump-cmm', '*.dump-cmm-*'
+      aliases 'cmm'
+
+      ws = %r(\s|//.*?\n|/[*](?:[^*]|(?:[*][^/]))*[*]+/)mx
+
+      # Make sure that this is not a preprocessor macro, e.g. `#if` or `#define`.
+      id = %r((?!#[a-zA-Z])[\w#\$%_']+)
+
+      complex_id = %r(
+        (?:[\w#$%_']|\(\)|\(,\)|\[\]|[0-9])*
+        (?:[\w#$%_']+)
+      )mx
+
+      state :root do
+        rule %r/\s+/m, Text
+
+        # sections markers
+        rule %r/^=====.*=====$/, Generic::Heading
+
+        # timestamps
+        rule %r/^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d+ UTC$/, Comment::Single
+
+        mixin :detect_section
+        mixin :preprocessor_macros
+
+        mixin :info_tbls
+        mixin :comments
+        mixin :literals
+        mixin :keywords
+        mixin :types
+        mixin :infos
+        mixin :names
+        mixin :operators
+
+        # escaped newline
+        rule %r/\\\n/, Comment::Preproc
```

Comment:
I had a look at a couple of lexers and it appears to me that this is typically tokenised as `Text`. What do you think about changing this one?

Reply:
Done :smiley: 

It was a "3 minute change". So, no worries ...

BTW If any questions regarding `ghc-cmm` and `ghc-core` arise later, please feel free to "ping" me. (Probably by mentioning `@supersven` in a comment.)

Reply:
Thanks! Will do!